python/nav/web/seeddb/utils/edit.py

```diff
@@ -99,9 +99,28 @@ def render_edit(
     }
     if obj:
         if obj.pk:
+            if verbose_name == "room":
+                detail_link = reverse_lazy('room-info', kwargs={'roomid': obj.pk})
+            elif verbose_name == "location":
+                detail_link = reverse_lazy(
+                    'location-info', kwargs={'locationid': obj.pk}
+                )
+            elif verbose_name == "vlan":
+                detail_link = reverse_lazy('vlan-details', kwargs={'vlanid': obj.pk})
+            elif verbose_name == "device group":
+                detail_link = reverse_lazy(
+                    'netbox-group-detail', kwargs={'groupid': obj.pk}
+                )
+            elif verbose_name == "prefix":
+                detail_link = reverse_lazy(
+                    'prefix-details', kwargs={'prefix_id': obj.pk}
+                )
```

Comment:
This feels hackish to me. At the very least, it should be factored out as a separate function. Preferably, each branch here should be pushed back to the individual page definitions, which is where you would expect to find the domain knowledge of each data type.